synapse/api/auth/msc3861_delegated.py

```diff
@@ -270,14 +289,14 @@ async def get_user_by_access_token(
             # XXX: This is a temporary solution so that the admin API can be called by
             # the OIDC provider. This will be removed once we have OIDC client
             # credentials grant support in matrix-authentication-service.
-            logging.info("Admin toked used")
+            logging.info("Admin token used")
             # XXX: that user doesn't exist and won't be provisioned.
             # This is mostly fine for admin calls, but we should also think about doing
             # requesters without a user_id.
             admin_user = UserID("__oidc_admin", self._hostname)
```

Comment:
I would really do the impersonation at this level. Why? Because it is the same workaround as this shared admin token, it will disappear once we have a proper design on how MAS interacts with Synapse, and it should not be used by any other token